src/net/udp.rs

```diff
@@ -86,6 +87,34 @@ impl UdpSocket {
         }
     }
 
+    /// Creates a future that will write the entire contents of the buffer `buf` to
+    /// the stream `a` provided.
+    ///
+    /// The returned future will return after data has been written to the outbound
+    /// socket. 
+    /// The future will resolve to the stream as well as the buffer (for reuse if
+    /// needed).
+    ///
+    /// Any error which happens during writing will cause both the stream and the
+    /// buffer to get destroyed.
+    ///
+    /// The `buf` parameter here only requires the `AsRef<[u8]>` trait, which should
+    /// be broadly applicable to accepting data which can be converted to a slice.
+    /// The `Window` struct is also available in this crate to provide a different
+    /// window into a slice if necessary.
+    pub fn send_dgram<'a, T>(&'a self, buf: T, addr : &'a SocketAddr) -> SendDGram<T>
```

Comment:
The name here to me is a little innocous. This is to UDP sockets as `tokio_core::io::write_all` is to TCP streams (modulo the partial write semantics). I wonder if we should name this `std::net::send_to` perhaps to be consistent?